File: .old/v4/engine/ai/bartender.lua
local Bartender = setmetatable({}, ai.NPC)
Bartender.__index = Bartender

function Bartender:new(x, y)
   return ai.NPC.new(self, x, y)
end

-------------
-- PRIVATE --
-------------

function Bartender:__tostring()
   return '[Bartender]'
end

function Bartender:type() return 'Bartender' end

return Bartender

-- vim: ts=3:sw=3:sts=3:expandtab

File: engine/npc/bartender.cc
#include "engine/npc/bartender.h"

#include <memory>
using namespace std;

#include "./main.h"
#include "ui/ui.h"
#include "ui/dialogmanager.h"

Bartender::Bartender(Point init, class City& city)
    : Worker(init, city, 0)
{
}

Bartender::~Bartender()
{
}


void
Bartender::TalkToHero()
{
    ui->Dialog().Bartender(*this);
}


string 
Bartender::News() const
{
    return _("No news is good news.");
}


// vim: ts=4:sw=4:sts=4:expandtab
